components/MainPage/SearchBox.js

```diff
@@ -337,6 +358,20 @@ class SearchBox extends React.Component {
     return menuItems;
   }
 
+  getCode(searchWithCode){
+      let code = null;
+      let codeJSON = JSON.parse(this.props.code);
```

Comment:
But isn't this an array?

Reply:
I guess @gregoirefora was using a user with only one role, and a badly configured keycloak returning and object instead of an array. Will fix it.

1. parsing roles
2. then using first role found with _r=""_ or _r="editStops"_ (i.e. a role allowing user to view or edit stops)
3. using org from that role

Sounds good to you ?